ethereum/core/src/main/java/org/hyperledger/besu/ethereum/core/Transaction.java

```diff
@@ -411,14 +416,21 @@ public Optional<BigInteger> getChainId() {
   @Override
   public Address getSender() {
```

Comment:
I think you could improve readability of this method this way:

```java
  @Override
  public Address getSender() {
    if (sender == null) {
      Optional<Address> cachedSender = Optional.ofNullable(senderCache.getIfPresent(getHash()));
      sender = cachedSender.orElseGet(this::computeSender);
    }
    return sender;
  }

  private Address computeSender() {
    final SECPPublicKey publicKey =
            signatureAlgorithm
                    .recoverPublicKeyFromSignature(getOrComputeSenderRecoveryHash(), signature)
                    .orElseThrow(
                            () ->
                                    new IllegalStateException(
                                            "Cannot recover public key from signature for " + this));
    Address calculatedSender = Address.extract(Hash.hash(publicKey.getEncodedBytes()));
    senderCache.put(this.hash, calculatedSender);
    return calculatedSender;
    return Address.extract(Hash.hash(publicKey.getEncodedBytes()));
  }
```

Reply:
Thanks @sebastien-perpignane, I will update the code with your suggestion